pkgs/development/libraries/harfbuzz/default.nix

```diff
@@ -77,29 +81,30 @@ stdenv.mkDerivation (finalAttrs: {
     libintl
     pkg-config
     python3
-    glib
     gtk-doc
     docbook-xsl-nons
     docbook_xml_dtd_43
-  ] ++ lib.optional withIntrospection gobject-introspection;
+  ] ++ lib.optional withIntrospection gobject-introspection
+  ++ lib.optional withGlib glib;
 
-  buildInputs = [ glib freetype ]
+  buildInputs = [ freetype ]
+    ++ lib.optionals withGlib [ glib ]
     ++ lib.optionals withCoreText [ ApplicationServices CoreText ];
 
   propagatedBuildInputs = lib.optional withGraphite2 graphite2
-    ++ lib.optionals withIcu [ icu harfbuzz ];
+    ++ lib.optionals withIcu [ icu ];
 
   doCheck = true;
 
   # Slightly hacky; some pkgs expect them in a single directory.
-  postFixup = lib.optionalString withIcu ''
-    rm "$out"/lib/libharfbuzz.* "$dev/lib/pkgconfig/harfbuzz.pc"
-    ln -s {'${harfbuzz.dev}',"$dev"}/lib/pkgconfig/harfbuzz.pc
-    ${lib.optionalString stdenv.isDarwin ''
-      ln -s {'${harfbuzz.out}',"$out"}/lib/libharfbuzz.dylib
-      ln -s {'${harfbuzz.out}',"$out"}/lib/libharfbuzz.0.dylib
-    ''}
-  '';
+  #postFixup = lib.optionalString withIcu ''
+  #  rm "$out"/lib/libharfbuzz.* "$dev/lib/pkgconfig/harfbuzz.pc"
+  #  ln -s {'${harfbuzz.dev}',"$dev"}/lib/pkgconfig/harfbuzz.pc
```

Comment:
I was encountering a recursion error on this line. Not sure the best way to resolve. I suspect that harfbuzz withIcu has been broken for some time.

Reply:
I don't think `harfbuzzFull` is broken on master. Probably just change `withIcu` to `(withIcu && !stdenv.hostPlatform.isMinGW)` here.

Reply:
Applied your recommend change.